rts/Sim/MoveTypes/GroundMoveType.h

```diff
@@ -127,7 +127,7 @@ class CGroundMoveType : public AMoveType
 	float3 GetObstacleAvoidanceDir(const float3& desiredDir);
 	float3 Here() const;
 
-	#define SQUARE(x) ((x) * (x))
+	#define SQUARE(x) ((x) * (abs(x)))
```

Comment:
This should probably not be a macro because that can recalculate `x` twice.
It probably also shouldn't be called `SQUARE` anymore (unsure if there's a concise math term for this, maybe just "signed square"?)

Reply:
I wasn't able to find a proper mathy term either, so I called it SignedSquare as you suggested and moved it to a function.